### PR TITLE
feat: reject unseal object after upload or replicate fail

### PR DESCRIPTION
**Pull request by joeylichang**
### Description

reject unseal object after upload or replicate fail

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 

N/A

### CODE REVIEW

File: base/gfspapp/sign_server.go [hunk 1]
```diff
@@ -41,6 +41,11 @@ func (g *GfSpBaseApp) GfSpSign(ctx context.Context, req *gfspserver.GfSpSignRequ
 		if err != nil {
 			log.CtxErrorw(ctx, "failed to seal object", "error", err)
 		}
+	case *gfspserver.GfSpSignRequest_RejectObjectInfo:
+		err = g.signer.RejectUnSealObject(ctx, t.RejectObjectInfo)
```
Comment (sysvm) [inline base/gfspapp/sign_server.go:45]: fix `RejectUnSealObject` as `RejectUnsealedObject` may be better.

File: base/gfspapp/sign_server.go [hunk 2]
```diff
@@ -41,6 +41,11 @@ func (g *GfSpBaseApp) GfSpSign(ctx context.Context, req *gfspserver.GfSpSignRequ
...
+		if err != nil {
+			log.CtxErrorw(ctx, "failed to reject unseal object", "error", err)
```
Comment (sysvm) [inline base/gfspapp/sign_server.go:47]: fix `unseal` as `unsealed` may be better.

File: base/gfspclient/signer.go [hunk 3]
```diff
@@ -78,6 +78,30 @@ func (s *GfSpClient) SealObject(ctx context.Context, object *storagetypes.MsgSea
 	return nil
 }
 
+func (s *GfSpClient) RejectUnSealObject(
+	ctx context.Context,
+	object *storagetypes.MsgRejectSealObject) error {
```
Comment (sysvm) [inline base/gfspclient/signer.go:83]: these parameters can be moved to one line.

File: modular/manager/manager.go [hunk 4]
```diff
@@ -26,6 +26,12 @@ const (
 
 	// DiscontinueBucketLimit define the max buckets to fetch in a single request
 	DiscontinueBucketLimit = int64(500)
+
+	// RejectUnSealObjectRetry defines the retry number of sending reject unseal object tx.
+	RejectUnSealObjectRetry = 3
```
Comment (sysvm) [inline modular/manager/manager.go:31]: fix `RejectUnSealObjectRetry` as `RejectUnsealedObjectRetry`

File: modular/manager/manager.go [hunk 5]
```diff
@@ -26,6 +26,12 @@ const (
...
+
+	// RejectUnSealObjectTimeout defines the timeout of sending reject unseal object tx.
+	RejectUnSealObjectTimeout = 3
```
Comment (sysvm) [inline modular/manager/manager.go:34]: fix `RejectUnSealObjectTimeout ` as `RejectUnsealedObjectTimeout`

File: modular/signer/signer_client.go [hunk 6]
```diff
@@ -159,7 +159,11 @@ func (client *GreenfieldChainSignClient) VerifySignature(scope SignType, msg, si
 }
 
 // SealObject seal the object on the greenfield chain.
-func (client *GreenfieldChainSignClient) SealObject(ctx context.Context, scope SignType, sealObject *storagetypes.MsgSealObject) ([]byte, error) {
+func (client *GreenfieldChainSignClient) SealObject(
+	ctx context.Context,
+	scope SignType,
+	sealObject *storagetypes.MsgSealObject) (
+	[]byte, error) {
```
Comment (sysvm) [inline modular/signer/signer_client.go:166]: move these params two one line.

File: base/gfspapp/sign_server.go (hunk 1 above)
Comment (joeylichang) [reply base/gfspapp/sign_server.go:45]: keep same to greenfield

File: base/gfspapp/sign_server.go (hunk 2 above)
Comment (joeylichang) [reply base/gfspapp/sign_server.go:47]: ditto

File: base/gfspclient/signer.go (hunk 3 above)
Comment (joeylichang) [reply base/gfspclient/signer.go:83]: fixed

File: modular/manager/manager.go (hunk 4 above)
Comment (joeylichang) [reply modular/manager/manager.go:31]: ditto

File: modular/manager/manager.go (hunk 5 above)
Comment (joeylichang) [reply modular/manager/manager.go:34]: ditto

File: modular/signer/signer_client.go (hunk 6 above)
Comment (joeylichang) [reply modular/signer/signer_client.go:166]: fixed

File: modular/signer/signer_client.go [hunk 7]
```diff
@@ -218,6 +222,59 @@ func (client *GreenfieldChainSignClient) SealObject(ctx context.Context, scope S
 	return txHash, nil
 }
 
+// RejectUnSealObject reject seal object on the greenfield chain.
+func (client *GreenfieldChainSignClient) RejectUnSealObject(
+	ctx context.Context,
+	scope SignType,
```
[the remark this answers is not in the document]
Comment (joeylichang) [reply modular/signer/signer_client.go:228]: fixed

File: modular/signer/signer_client.go [hunk 8]
```diff
@@ -218,6 +222,59 @@ func (client *GreenfieldChainSignClient) SealObject(ctx context.Context, scope S
...
+
+	msgRejectUnSealObject := storagetypes.NewMsgRejectUnsealedObject(km.GetAddr(), rejectObject.GetBucketName(), rejectObject.GetObjectName())
+	mode := tx.BroadcastMode_BROADCAST_MODE_ASYNC
+	txOpt := &ctypes.TxOption{
+		Mode:     &mode,
+		GasLimit: client.gasLimit,
+		Nonce:    nonce,
+	}
+	resp, err := client.greenfieldClients[scope].BroadcastTx(ctx, []sdk.Msg{msgRejectUnSealObject}, txOpt)
+
```
[the remark this answers is not in the document]
Comment (joeylichang) [reply modular/signer/signer_client.go:249]: logical partition

[the remark this answers is not in the document]
Comment (sysvm) [reply modular/signer/signer_client.go:249]: but shouldn't use in checking err

File: base/gfspapp/sign_server.go (hunk 1 above)
Comment (sysvm) [reply base/gfspapp/sign_server.go:45]: greenfield used `NewMsgRejectUnsealedObject`

**Review by will-2012 — APPROVED** [review]

**Pull request merged**